src/dragonball/src/device_manager/mod.rs

```diff
@@ -591,14 +591,25 @@ impl DeviceManager {
             .map_err(|_| StartMicroVmError::EventFd)?;
 
         info!(self.logger, "init console path: {:?}", com1_sock_path);
-        if let Some(path) = com1_sock_path {
-            if let Some(legacy_manager) = self.legacy_manager.as_ref() {
-                let com1 = legacy_manager.get_com1_serial();
-                self.con_manager
-                    .create_socket_console(com1, path)
-                    .map_err(StartMicroVmError::DeviceManager)?;
+        match com1_sock_path {
+            Some(path) => {
+                if path == "None" {
```

Comment:
Why do you need to check if this is `"None"`?
Why can't this case be covered by `match com1_sock_path` which `com1_sock_path` is `None`?

Reply:
This case can be seen in PR: [Create VM from command line arguments](https://github.com/openanolis/dbs-cli/pull/1). As described in the last comment, I must set a unique value to make a distinction.

Reply:
```rust
            // as in crate `dragonball` serial_path will be assigned with a default value,
            // we need a special token to enable the stdio console.
            serial_path: Some("None".to_owned()),
```

Reply:
That's strange. Could you just set it to `None` instead of string `"None"`?

Reply:
> ```rust
> if let some(path) = com1_sock_path {
>     // The token "None" is used to create stdio console. (means not to create sock console)"
>     if path != "None" {
>         let com1 = legacy_manager.get_com1_serial();
>         self.con_manager
>             .create_socket_console(com1, path)
>             .map_err(StartMicroVmError::DeviceManager)?;
>         return Ok(());
>     }
> }
> ```

If `None` is given, we would always get a default value, which is done by this code snippet. It ok to set it to `None`, but the default value of serial_path is unexpected.

Reply:
Why default value is unexpected?
Do you mean the default value could not work to create a socket console?

Reply:
When it is time to create a stdio console, we don't want to create a socket console. Similarly, it is not supposed to create a stdio console when we need a socket console. The default value makes these two separate operations bound together, therefore we must set a special value to distinguish them. To make it clear, I change the `"None"` to `"stdio"`.

```rust
if let Some(legacy_manager) = self.legacy_manager.as_ref() {
    if let Some(path) = com1_sock_path {
        if "stdio" != path {
            let com1 = legacy_manager.get_com1_serial();
            self.con_manager
                .create_socket_console(com1, path)
                .map_err(StartMicroVmError::DeviceManager)?;
            return Ok(());
        }
    }

    let com1 = legacy_manager.get_com1_serial();
    self.con_manager
        .create_stdio_console(com1)
        .map_err(StartMicroVmError::DeviceManager)?;
}
```

Reply:
By the way,  it seems that we cannot remove the functionality, setting a default value to serial path.

Reply:
> Why default value is unexpected? Do you mean the default value could not work to create a socket console?

With the default value, `device_manager` will create two consoles (socket console and stdio console) simultaneously, which is unexpected.